tutorials/time-series-modeling/Lab_05.md

```diff
@@ -0,0 +1,392 @@
+  image.select(harmonicIndependents)   
+  .multiply(harmonicTrendCoefficients)   
+  .reduce('sum')   
+  .rename('fitted'));
+});
+// Plot the fitted model and the original data at the ROI.
+print(ui.Chart.image.series(fittedHarmonic.select(['fitted','NDVI']), roi,
+                  ee.Reducer.mean(), 30)  
+  .setSeriesNames(['NDVI', 'fitted'])  
+  .setOptions({   
+  title: 'Harmonic Model: Original and Fitted Values',   
+  lineWidth: 1,   
+  pointSize: 3,}));
+```
+
+The harmonic model (red data points) does an adequate job of predicting the data - There is a datapoint in Feb 2021 that is significantly lower, but this appears to be an outlier. Additionally, the model misses a significant dip in July 2015, although this might be due to irregular climate conditions - other years did not have the same dip.  
+
+> **Question 1**: Upload the resulting graphic and interpret it.
+
+Although any coefficients can be mapped directly, it is often useful and interesting to map the phase and amplitude of the estimated harmonic model. First, compute phase and amplitude from the coefficients, then incorporate this information into each pixel. Use inspector to look at the pixels and note their phase and amplitude.
+
+
+```javascript
+// Compute phase and amplitude.
+var phase = harmonicTrendCoefficients.select('sin')
+		.atan2(harmonicTrendCoefficients.select('cos'))  
+		// Scale to [0, 1] from radians.  
+  .unitScale(-Math.PI, Math.PI);
+var amplitude = harmonicTrendCoefficients.select('sin')
+		.hypot(harmonicTrendCoefficients.select('cos'))  
+		// Add a scale factor for visualization.  
+  .multiply(5);
+// Compute the mean NDVI.
+var meanNdvi = filteredLandsat.select('NDVI').mean();
+// Use the HSV to RGB transformation to display phase and amplitude.
+var rgb = ee.Image.cat([
+                        phase,
+                        amplitude, 
+                        meanNdvi
+		]).hsvToRgb()
+Map.addLayer(rgb, {}, 'phase (hue), amplitude (sat), ndvi (val)');
+```
+
+> **Question 2**: Upload the resulting map layer and describe its salient features.
+
+---
+
+###  Complex Time Series Modeling
+
+A time series can be decomposed as the sum of sinusoids at different frequencies. The harmonic model presented here can be extended by adding bands that represent higher frequencies and the corresponding `sin()` band for a harmonic component to account for two cycles per year.  
+
+You can look at this GEE [example](https://code.earthengine.google.com/2669122497313113fc4bb81bc8352828) of using multiple sinusoids to build a more complex harmonic model. Note that each year there is a high peak in June and a secondary peak in January - this harmonic model consisting of two sinusoids with separate frequencies and amplitudes is able to account for that. However, the error values in this model are high and the fit is quite inexact. We can see extreme drops in the NDVI value that the model misses, and several peaks each year that do not fit.
```

Comment:
Use updated script link that user Landsat 8 Collection 2 SR
https://code.earthengine.google.com/bc9e4410bc87af635be52501ed288b0e

Reply:
Updated script link with collection 2